packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -915,13 +915,26 @@ class IntelliJValidatorOnMac extends IntelliJValidator {
     }
 
     final List<String> split = version.split('.');
-
     if (split.length < 2) {
       return null;
     }
+    final int major = int.parse(split[0]);
+    final int minor = int.parse(split[1]);
+
+    if (major >= 2020) {
+      _pluginsPath = globals.fs.path.join(
+        globals.fsUtils.homeDirPath,
+        'Library',
+        'Application Support',
+        'JetBrains',
+        '$id$major.$minor',
+        'plugins',
+      );
+      if (globals.fs.isDirectorySync(_pluginsPath)) {
+        return _pluginsPath;
+      }
+    }
```

Comment:
This sets `_pluginsPath` then immediately resets it to the legacy location if that directory isn't found.  How about something like:
```dart
    final String homeDirPath = globals.fsUtils.homeDirPath;
    String pluginsPath = globals.fs.path.join(
      homeDirPath,
      'Library',
      'Application Support',
      'JetBrains',
      '$id$major.$minor',
      'plugins',
    );

    // Fallback to legacy location from < 2020.
    if (!globals.fs.isDirectorySync(pluginsPath)) {
      pluginsPath = globals.fs.path.join(
        homeDirPath,
        'Library',
        'Application Support',
        '$id$major.$minor',
      );
    }
    _pluginsPath = pluginsPath;

    return _pluginsPath;
```

This is two I/O checks instead of one, though I think that's what we do for spots like https://github.com/flutter/flutter/blob/c2b7342ca470b11cfaad4fbfb094f73aa4c85320/packages/flutter_tools/lib/src/android/android_sdk.dart#L286-L294.

Reply:
Added via 6205755